lib/Benchmark.t

```diff
@@ -40,6 +40,73 @@ sub cmp_delta {
     return $max/$min <= (1+$delta);
 }
 
+sub splatter {
+    my ($message) = @_;
+    my $splatter = <<~'EOF_SPLATTER';
+    Please file a ticket to report this. Our bug tracker can be found at
+
+        https://github.com/Perl/perl5/issues
+
+    Make sure you include the full output of perl -V, also uname -a,
+    and the version details for the C compiler you are using are
+    very helpful.
+
+    Please also try compiling and running the C program that can
+    be found at
+
+        https://github.com/Perl/perl5/issues/20839#issuecomment-1439286875
+
+    and provide the results (or compile errors) as part of your
+    bug report.
+
+    EOF_SPLATTER
+
+    if ( $message =~ s/\.\.\.//) {
+        $splatter =~ s/Please/please/;
+    }
+    die $message, $splatter;
+}
+
+{
+    # Benchmark may end up "looping forever" if time() or times() are
+    # broken such that they do return changing values over time. The
+    # following crude test is intended to ensure that we can reply on
+    # them sufficiently we don't infinite loop.
+
+    my @times0;
+    for (1..3) {
+        my $end_time = time + 1;
+        my $count= 0;
+        my $scale = $ENV{FAST_CPU} || 1;
+        my $count_threshold = 100_000;
+        while ( time < $end_time ) {
```

Comment:
In the case where time does not work here (forcing while (1)), this entire block from for (1..3) { ... } takes 2 minutes and 3 seconds on my pretty good hardware. I suppose this should be rare, but it will still "look" like hanging. I wonder if it can be adjusted down, but maybe it doesn't matter.

Reply:
Im open to pushing it down. This was purely a finger in the air kind of estimate. So ill try out a lower number.

I do think at a certain level it doesn't matter. The main objective is to error eventually, instead of never. And so far we haven't seen any broken time() functions afaik, just times(), this was just a sanity defense if there /was/ a time out there so that our sanity check didn't recreate the bug it was supposed to protect against. :-(

But we can try a lower number I think.  Ill queue something up in the next few minutes.

Reply:
And also done. I added some ENV vars to make it easier to simulate time() or times() failing. 

```
+    #   time()    -> SIMULATE_BROKEN_TIME_FUNCTION
+    #   times()   -> SIMULATE_BROKEN_TIMES_FUNCTION
```

Reply:
Nice